app/models/generic_object.rb

```diff
@@ -84,4 +94,21 @@ def property_setter(name, value)
 
     self.properties = properties.merge(name => val)
   end
+
+  # Suppose the methods are located under /Generic/Methods
+  # go.my_host(:user_id=>u.id)
+  def call_automate(method_name, options = {}, q_options = {})
+    raise "user_id is not specified in method [#{method_name}]" unless options[:user_id]
+
+    options.reverse_merge!(
```

Comment:
The user should not specify the namespace/class_name values.  We should pass the call to automate into a new instance under `/System/Process` named `GenericObject` which would use substitution to drive the method to a new class.

The substitution would use the generic object definition name and the method name to resolve to the method to run.

/System/Process/GenericObject would contain a relationship like: `/GenericObject/<generic_object_definition>/<method_name>`

I would suggest moving the actual logic to call automate into a separate PR and use this PR for support defining methods on the generic object definition and the placeholder method for initiating the call to automate.